### PR TITLE
Fix/storybook typecheck step

### LINKED ISSUE #162: CI Storybook typescript typecheck step
We need a typescript typecheck step for storybook files in order to break the CI when a storybook file is incorrect.

The [Storybook Typescript documentation](https://storybook.js.org/docs/react/configure/typescript#mainjs-configuration) describes how to enable type checking.

### Tasks
- [ ] enable type checking in the storybook config
- [ ] enable eslint for storybook (also described in the linked doc)
- [ ] use correct tsconfig.json (the config where storybook resides)
- [ ] figured out the correct value for the `async` setting. It maybe worth while to set this to `false`.
- [ ] check if any other settings need to be enabled.

### PR DESCRIPTION
**Pull request by DanielvanVliet**
This is a different approach than was described in the issue, since I had trouble getting that to work on our multi-component project.

Added a typescript tsconfig file that scans all *.stories.tsx files and typechecks them.
As a result of this I had to fix type-errors and create new issues to solve problems in *.stories.tsx.

One notable change is how the button property in ListItem was changed back to any, this will be solved in a future PR #169 when list api is updated.

**Closing issues**

closes #162

### CODE REVIEW

File: src/components/FormControlLabel/src/FormControlLabel.stories.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default {
 } as Meta;
 
 const Template: Story<FormControlLabelProps> = (args: FormControlLabelProps) => (
-  <FormControlLabel label="Checkbox" control={<Checkbox {...args} />} {...args} />
+  <FormControlLabel {...args} label="Checkbox" control={<Checkbox {...args} />} />
```
Comment (GewoonMaarten) [inline src/components/FormControlLabel/src/FormControlLabel.stories.tsx:22]: Why was this ordering changed?

File: src/components/Select/src/Select.stories.tsx [hunk 2]
```diff
@@ -34,6 +34,7 @@ export function DefaultSelect(props: SelectProps): React.ReactElement {
 
   return (
     <FormControl className={classes.formControl}>
+      {/*  */}
```
Comment (GewoonMaarten) [inline src/components/Select/src/Select.stories.tsx:37]: Is this comment necessary?

**Review by GewoonMaarten — CHANGES_REQUESTED** [review]

File: src/components/FormControlLabel/src/FormControlLabel.stories.tsx (hunk 1 above)
Comment (DanielvanVliet) [reply src/components/FormControlLabel/src/FormControlLabel.stories.tsx:22]: If left as before, typescript considers label and control to be specified twice (once in ...args) and gives an error.
Swapping them like this causes label and control to 'override' the one in {...args} and no error occurs.

File: src/components/Select/src/Select.stories.tsx (hunk 2 above)
Comment (DanielvanVliet) [reply src/components/Select/src/Select.stories.tsx:37]: nope fixed

**Review by GewoonMaarten — APPROVED** [review]

**Review by mjcarsjens — CHANGES_REQUESTED** [review]
Solid addition, let's not merge this until the issues which occur right now are fixed, but afterwards fine by me!

File: src/components/List/src/ListItem.tsx [hunk 3]
```diff
@@ -19,7 +19,8 @@ export interface ListItemProps extends BaseProps {
    * If `true`, the list item will be a button (using `ButtonBase`). Props intended
    * for `ButtonBase` can then be applied to `ListItem`.
    */
-  button?: false;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  button?: any;
```
Comment (GewoonMaarten) [inline src/components/List/src/ListItem.tsx:23]: This sucks, but I assume their is no other type that fits?

**Review by GewoonMaarten — DISMISSED** [review]

File: src/components/List/src/ListItem.tsx (hunk 3 above)
Comment (DanielvanVliet) [reply src/components/List/src/ListItem.tsx:23]: It is a temporary fix, since when list-design is merged this property won't  be exposed anymore.

File: .gitattributes [hunk 4]
```diff
@@ -0,0 +1 @@
+* text=auto eol=lf
```
Comment (bddjong) [inline .gitattributes:1]: no newline

File: src/components/Tab/src/Tab.stories.tsx [hunk 5]
```diff
@@ -23,11 +23,11 @@ const Template: Story<TabProps> = (args: TabProps) => {
     <TabContext value={value.toString()}>
       <Tabs
         value={value}
-        onChange={(event: React.ChangeEvent<unknown>, newValue: number) => {
+        onChange={(_event: React.ChangeEvent<unknown>, newValue: number) => {
```
Comment (bddjong) [inline src/components/Tab/src/Tab.stories.tsx:26]: Why was this renamed?

Comment (DanielvanVliet) [reply src/components/Tab/src/Tab.stories.tsx:26]: To fix a typecheck error:
err
Eslint provided a quickfix suggesting this solution, to signal to other developers that the parameter is unused in the function.

**Review by GewoonMaarten — DISMISSED** [review]

**Review by mjcarsjens — APPROVED** [review]
